frontend/packages/helm-plugin/src/components/details-page/HelmReleaseDetails.tsx

```diff
@@ -154,7 +153,10 @@ const HelmReleaseDetails: React.FC<HelmReleaseDetailsProps> = ({ secret, match }
     return () => {
       ignore = true;
     };
-  }, [helmReleaseName, namespace]);
+    // On upgrading/rolling back to another version a new helm release is created.
+    // For fetching and showing the details of the new release adding secret.data as depedency here
+    // since secret's data list gets updated when a new release is created.
+  }, [helmReleaseName, namespace, secret.data]);
```

Comment:
if a new secret is created every time a release is rolled back or upgraded then it would be better to use `secret.data.length`.

Reply:
Okay, just tried this locally. It seems there's some issue where the page gets stuck while doing a rollback if we use `secret.data.length`. So its okay as it is now.